node-tests/blueprints/route-test.js

```diff
@@ -393,6 +394,293 @@ describe('Blueprint: route', function() {
     });
   });
 
+  describe('in app - module unification', function() {
+    beforeEach(function() {
+      return emberNew()
+        .then(() => generateFakePackageManifest('ember-cli-qunit', '4.1.0'))
+        .then(() => {
+          fs.ensureDirSync('src');
+          fs.writeFileSync('src/router.js', fs.readFileSync('app/router.js'));
+        });
+    });
+
+    it('route foo', function() {
+      return emberGenerateDestroy(['route', 'foo'], _file => {
+        expect(_file('src/ui/routes/foo/route.ts'))
+          .to.contain("import Route from '@ember/routing/route';")
+          .to.contain('export default class Foo extends Route.extend({')
```

Comment:
Guidance from those working on ES6 class support is that this pattern is "discouraged", although in e-cli-ts it's all over the place. We likely need to figure out a way to address this soon
https://github.com/emberjs/ember.js/pull/17174#discussion_r230233384

At the very least, once ember 3.6 lands we'll want to update blueprints to align with